src/fluree/db/virtual_graph.cljc

```diff
@@ -0,0 +1,36 @@
+(ns fluree.db.virtual-graph
+  (:require [clojure.string :as str]
+            [fluree.db.json-ld.iri :as iri]))
+
+(defprotocol UpdatableVirtualGraph
+  (upsert [this source-db new-flakes remove-flakes] "Updates the virtual graph with the provided flakes. Returns async chan with new updated VirtualGraph or exception.")
+  (initialize [this source-db] "Initialize a new virtual graph based on the provided db - returns promise chan of eventual result"))
+
+(defn vg-type-name
+  [vg]
+  (-> vg :type first iri/get-name str/lower-case))
+
+(defn vg-type-kw
+  [vg]
+  (keyword (vg-type-name vg)))
+
+(defmulti write-vg
+          (fn [_index-catalog vg]
+            (vg-type-kw vg)))
+
+(defn storage-path
+  [vg-type-kw db-alias vg-alias]
+  (str/join "/" [db-alias (name vg-type-kw) vg-alias]))
```

Comment:
Why not just use `vg-type-name` here instead of taking that value, turning it into the keyword and then back into a string?

Reply:
In this case it is an internal VG type that is being used for a subdirectory... e.g. :bm25, :hnsw, etc. 

I renamed the arg keyword to just `vg-type` to minimize confusion with the function names above which this unintentionally was identical to.

vg-type-name is used when parsing original JSON-LD to extract a vg type.